bgpd/bgp_route.c

```diff
@@ -1990,6 +1993,27 @@ bool subgroup_announce_check(struct bgp_dest *dest, struct bgp_path_info *pi,
 				peer->host, prefix2str(p, buf, sizeof(buf)));
 
 			bgp_attr_flush(attr);
+
+			/* TBD : Not sure if this is the correct way to fetch
+			 * peer from group.
+			 * Notify BGP Conditional advertisement scanner process.
+			 */
+			if (ADVERTISE_MAP_NAME(filter)
+			    || CONDITION_MAP_NAME(filter)) {
+				struct peer *temp_peer;
+				struct listnode *temp_node, *temp_nnode = NULL;
+				for (ALL_LIST_ELEMENTS(bgp->peer, temp_node,
+						       temp_nnode, temp_peer)) {
+					if (!CHECK_FLAG(peer->flags,
+							PEER_FLAG_CONFIG_NODE))
+						continue;
+					if (strcmp(peer->host, temp_peer->host)
```

Comment:
I think this should be handled per subgroup and not per peer.

Reply:
advertise-map is per peer context so scanning happens per peer.
And this check is needed when we configure route-map outbound along with advertise-map and routes matching with condition-map/advertise-map are filtered out by route-map.
Configuring Route-map out and advertise-map is a valid scenario ?

Reply:
> Configuring Route-map out and advertise-map is a valid scenario ?

I think it is because you can have `rmap1` permitting 192.168.0.0/24, 192.168.1.0/24, and `advrmap1` advertising 192.168.1.0/24 only under conditions.

Reply:
Can you create such a setup for topotest?

Reply:
By the way, isn't enough to compare `peer == temp_peer`? or using `peer_cmp()` to make sure they are not in the same group, etc.

Reply:
addressed this comment. 
used peer == temp_peer. added test cases in topotest.